src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -342,6 +346,14 @@ protected static bool ItemspecContainsASingleItemReference(ItemSpec<P, I> itemSp
                     return false;
                 }
 
+                // If the itemSpec is a single call to an item function, like @(X->Something(...)), it may get this
+                // far, but shouldn't be treated as a single reference: the item function may return entirely
+                // different results from a bare reference like @(X).
+                if (itemExpressionFragment.Capture.Captures is object)
+                {
+                    return false;
```

Comment:
Should this always return false or should it occasionally return true? I think this is only used for performance, so it doesn't matter if it has false negatives now, but I'd like to see if I understand it correctly. Also could be relevant if we start using it elsewhere.

Reply:
It should always return `false`. The question this method answers is "can I just remove/update every item in the group, or do I need to expand the value to match against existing items?"

Renamed it for a bit more clarity there.